src/librustc_lint/builtin.rs

```diff
@@ -226,23 +226,30 @@ impl UnsafeCode {
     }
 }
 
-impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnsafeCode {
-    fn check_expr(&mut self, cx: &LateContext, e: &hir::Expr) {
-        if let hir::ExprKind::Block(ref blk, _) = e.node {
+impl EarlyLintPass for UnsafeCode {
+    fn check_attribute(&mut self, cx: &EarlyContext, attr: &ast::Attribute) {
+        if attr.check_name("allow_internal_unsafe") {
+            self.report_unsafe(cx, attr.span, "cannot use `allow_internal_unsafe` \
```

Comment:
I think we should sync the error message with the others. What do you think about "marking a macro definition with `allow_internal_unsafe`"?

Reply:
I restored this message, OK?